src/aiotaskq/concurrency_manager.py

```diff
@@ -0,0 +1,51 @@
+from functools import cached_property
+import logging
+import multiprocessing
+import os
+import typing as t
+
+from .exceptions import ConcurrencyTypeNotSupported
+from .interfaces import ConcurrencyType, IConcurrencyManager, IProcess
+
+
+class ConcurrencyManager:
+    """The user-facing facade for creating the right concurrency manager implementation."""
+
+    _instance: "IConcurrencyManager"
+
+    @classmethod
+    def get(cls, concurrency_type: str, concurrency) -> IConcurrencyManager:
+        if cls._instance:
+            return cls._instance
+        if concurrency_type == ConcurrencyType.MULTIPROCESSING:
+            cls._instance = MultiProcessing(concurrency=concurrency)
+            return cls._instance
+        raise ConcurrencyTypeNotSupported(
+            f'Concurrency type "{concurrency_type}" is not yet supported.'
+        )
+
+
+class MultiProcessing:
+    """Implementation of a ConcurrencyManager that uses the `multiprocess` built-in module."""
+
+    def __init__(self, concurrency: int) -> None:
+        self.concurrency = concurrency
+        self.processes: dict[int, IProcess] = {}
```

Comment:
Good catch. I was using it for a temporary implementation that I eventually threw away so know we don't need it, so I'll remove it.

Btw, by looking at the implementation, it seems like `pid` should already be set once we've called `start`:

```python
# multiprocessing.process.BaseProcess

    def start(self):
        '''
        Start child process
        '''
        self._check_closed()
        assert self._popen is None, 'cannot start a process twice'
        assert self._parent_pid == os.getpid(), \
               'can only start a process object created by current process'
        assert not _current_process._config.get('daemon'), \
               'daemonic processes are not allowed to have children'
        _cleanup()
        self._popen = self._Popen(self)

    ...

    @property
    def ident(self):
        '''
        Return identifier (PID) of process or `None` if it has yet to start
        '''
        self._check_closed()
        if self is _current_process:
            return os.getpid()
        else:
            return self._popen and self._popen.pid

    pid = ident
```
I have also confirmed this locally by print `proc.pid` before and after calling `proc.start()`. So in the future if we need to use pid for some process we know we can rely on it. But for now since we're not using it I'll remove it.